kotlin/perform-valve-isolation-trace/src/main/java/com/esri/arcgisruntime/sample/performvalveisolationtrace/MainActivity.kt

```diff
@@ -0,0 +1,296 @@
+    val deviceLayer = FeatureLayer(deviceFeatureTable)
+
+    // add the feature layers to the map
+    map.operationalLayers.addAll(arrayOf(distributionLineLayer, deviceLayer))
+
+    mapView.map = map
+
+    // make sure the fab doesn't obscure the attribution bar
+    mapView.addAttributionViewLayoutChangeListener { _, _, _, _, bottom, _, _, _, oldBottom ->
+      val layoutParams = fab.layoutParams as CoordinatorLayout.LayoutParams
+      layoutParams.bottomMargin += bottom - oldBottom
+    }
+
+    // show the options sheet when the floating action button is clicked
+    fab.setOnClickListener {
+      fab.isExpanded = !fab.isExpanded
+    }
+
+    // close the options sheet when the map is tapped
+    mapView.onTouchListener = object : DefaultMapViewOnTouchListener(this, mapView) {
+      override fun onTouch(view: View?, event: MotionEvent?): Boolean {
+        if (fab.isExpanded) {
+          fab.isExpanded = false
+        }
+        return super.onTouch(view, event)
+      }
+    }
+
+    // create and load the utility network
+    map.addDoneLoadingListener {
+      createUtilityNetwork()
+    }
+  }
+
+  /**
+   * Creates a utility network from the string resource url and loads it.
+   * Initializes the starting point and UI.
+   */
+  private fun createUtilityNetwork(
+  ) {
+    // create a graphics overlay for the starting location and add it to the map view
+    val startingLocationGraphicsOverlay = GraphicsOverlay().also {
+      mapView.graphicsOverlays.add(it)
+    }
+
+    // create and apply renderers for the starting point graphics overlay
+    val startingPointSymbol = SimpleMarkerSymbol(
+      SimpleMarkerSymbol.Style.CROSS,
+      Color.GREEN,
+      25f
+    )
+    startingLocationGraphicsOverlay.renderer = SimpleRenderer(startingPointSymbol)
+
+    // create a utility network from the url and load it
+    val utilityNetwork = UtilityNetwork(getString(R.string.utility_network_url), mapView.map)
+    utilityNetwork.loadAsync()
+    utilityNetwork.addDoneLoadingListener {
+      if (utilityNetwork.loadStatus == LoadStatus.LOADED) {
+        // get a trace configuration from a tier
+        val networkDefinition = utilityNetwork.definition
+        val domainNetwork = networkDefinition.getDomainNetwork("Pipeline")
+        val tier = domainNetwork.getTier("Pipe Distribution System")
+        val traceConfiguration = tier.traceConfiguration
+
+        // create a trace filter
+        traceConfiguration.filter = UtilityTraceFilter()
+
+        // get a default starting location
+        val networkSource = networkDefinition.getNetworkSource("Gas Device")
+        val assetGroup = networkSource.getAssetGroup("Meter")
+        val assetType = assetGroup.getAssetType("Customer")
+        val startingLocation = utilityNetwork.createElement(
+          assetType,
+          UUID.fromString("98A06E95-70BE-43E7-91B7-E34C9D3CB9FF")
+        )
+
+        // get the first feature for the starting location, and get its geometry
```

Comment:
this line isn't getting the first, it's getting. listenable future list

Reply:
What's a better way to word this? This whole block is getting only the starting location and only the first feature of that starting location to create a graphic. Is it better then to change this to "get a list of features" and clarify inside?

Reply:
oh, you've cleaned it up below. I guess the new comment adds the necessary context